jenkins/shared-library/vars/publish.groovy

```diff
@@ -54,15 +52,23 @@ void npmPublisher(Map config, String phase) {
 }
 
 void pythonPublisher(Map config, String phase) {
-	if (config.publisher == 'pypi' && phase == 'release') {
-		runScript('python3 setup.py bdist_wheel')
-		withCredentials([
-			usernamePassword(
-			credentialsId: PYTHON_CREDENTIALS_ID,
+	if (config.publisher != 'pypi') {
+		return
+	}
+
+	credentialsId = PYTHON_CREDENTIALS_ID
+	if (phase == 'alpha') {
+		credentialsId = TEST_PYTHON_CREDENTIALS_ID
+	}
+
+	String scriptFullFilepath = fileHelper.copyToTempFile(getResource('artifacts/scripts/pypi-publish.sh'), 'scripts/pypi-publish.sh')
+	withCredentials([
+		usernamePassword(
+			credentialsId: credentialsId,
 			usernameVariable: 'TWINE_USERNAME',
-			passwordVariable: 'TWINE_PASSWORD')]) {
-				runScript('python3 -m twine upload --repository pypi dist/*')
-		}
+			passwordVariable: 'POETRY_PYPI_TOKEN_PYPI')
+	]) {
```

Comment:
as discussed, should change this to use single credential (like npm)

Reply:
Updated the pypi creds to be secret text.  You should update them to be like npm